lib/errgonomic.rb

```diff
@@ -97,6 +150,10 @@ def expect!(msg)
       end
 
       # Return the inner value of an Err, else raise an exception when Ok.
+      #
+      # @example
+      #   Ok("foo").unwrap_err! # => raise Errgonomic::UnwrapError, "value is an Ok"
```

Comment:
From interface design perspective with errors: I think that beyond knowing the value was "ok" in this scenario they would also want to know what it held. I.e. if it's surprising that it's Ok, the contents might yield some clues.

Reply:
Nice. I haven't used this in Rust yet so was a bit on autopilot in the implementation. That makes a lot of sense.

Reply:
I'm not 100% sure what methods hide the original error message in Rust, I think some do. In general though, I consider exceptions a time when I want maximum context as by definition: something I didn't expect is happening.

Reply:
Added that, thanks!